test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndChannelLeakTest.java

```diff
@@ -0,0 +1,272 @@
+
+        //Create a writer.
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter("test1", new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+
+        @Cleanup
+        ReaderGroupManager groupManager = new ReaderGroupManagerImpl("test", controller, clientFactory,
+                connectionFactory);
+        ReaderGroup readerGroup = groupManager.createReaderGroup("reader", ReaderGroupConfig.builder().disableAutomaticCheckpoints().
+                build(), Collections.singleton("test1"));
+
+        //create a reader.
+        @Cleanup
+        EventStreamReader<String> reader_1 = clientFactory.createReader("readerId1", "reader", new JavaSerializer<>(),
+                ReaderConfig.builder().build());
+
+        //Write an event.
+        writer.writeEvent("0", "zero").get();
+        System.out.println("===> 2 Active Channel Count: " + connectionFactory.getActiveChannelCount());
+
+        //Read an event.
+        EventRead<String> event = reader_1.readNextEvent(10000);
+        assertNotNull(event);
+        assertEquals("zero", event.getEvent());
+        //this is +1 the socket count.
+        System.out.println("===> 3 Active Channel Count: " + connectionFactory.getActiveChannelCount());
+
+        // scale
+        Stream stream = new StreamImpl("test", "test1");
+        Map<Double, Double> map = new HashMap<>();
+        map.put(0.0, 0.33);
+        map.put(0.33, 0.66);
+        map.put(0.66, 1.0);
+        Boolean result = controller.scaleStream(stream, Collections.singletonList(0), map, executor).getFuture().get();
+        assertTrue(result);
+        //same as 3
+        System.out.println("===> 4 Active Channel Count: " + connectionFactory.getActiveChannelCount());
+
+        //Write more events.
+        writer.writeEvent("0", "one").get();
+        writer.writeEvent("0", "two").get();
+        writer.writeEvent("1", "three").get();
+
+        System.out.println("===> 5 Active Channel Count: " + connectionFactory.getActiveChannelCount());
+
+        //Add a new reader
+        @Cleanup
+        EventStreamReader<String> reader_2 = clientFactory.createReader("readerId2", "reader", new JavaSerializer<>(),
+                ReaderConfig.builder().build());
+        System.out.println("===> 6 Active Channel Count: " + connectionFactory.getActiveChannelCount());
+
+        event = reader_1.readNextEvent(10000);
+        assertNotNull(event);
+        System.out.println("===> Data read by reader_1 is " + event.getEvent());
+        System.out.println("===> 7 Active Channel Count: " + connectionFactory.getActiveChannelCount());
+
+        CompletableFuture<Checkpoint> chkPointResult = readerGroup.initiateCheckpoint("chk1", executor); //create a check point to ensure rebalance of readers
+        assertTrue(reader_2.readNextEvent(10000).isCheckpoint());
+        TimeUnit.SECONDS.sleep(30);
```

Comment:
No sleeps in unit tests. Setup a future so it only blocks for exactly as long as needed.